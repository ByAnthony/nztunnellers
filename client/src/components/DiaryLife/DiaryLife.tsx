import STYLES from '../ProfileDiary/ProfileDiary.module.scss';

type props = {
    maritalStatus: string | null,
    wife: string | null,
}

export function DiaryLife({ maritalStatus, wife }: props) {
  const displayLife = (status: string | null, wifeName: string | null) => {
    if (maritalStatus !== null && wifeName !== null) {
      return (
        <>
          <div className={STYLES['fullwidth-main-card']}>
            Life
          </div>
          <div className={STYLES['halfwidth-cards-container']}>
            <div className={STYLES['halfwidth-secondary-card']}>
              <div className={STYLES['halfwidth-secondary-card-title']}><p>Marital Status</p></div>
              <div><span>{ status }</span></div>
            </div>
            <div className={STYLES['halfwidth-secondary-card']}>
              <div className={STYLES['halfwidth-secondary-card-title']}><p>Wife</p></div>
              <div><span>{ wifeName }</span></div>
            </div>
          </div>
        </>
      );
    }

    const displayMaritalStatus = (statusWithoutWife: string | null) => (
      <div className={STYLES['halfwidth-cards-container']}>
        <div className={STYLES['halfwidth-main-card']}>
          <span>Life</span>
        </div>
        <div className={STYLES['halfwidth-secondary-card']}>
          <div className={STYLES['halfwidth-secondary-card-title']}><p>Marital Status</p></div>
          <div><span>{ statusWithoutWife }</span></div>
        </div>
      </div>
    );

    if (maritalStatus !== null && wifeName === null) {
      return displayMaritalStatus(maritalStatus);
    }
    return null;
  };

  return (
    <>
      { displayLife(maritalStatus, wife) }
    </>
  );
}
