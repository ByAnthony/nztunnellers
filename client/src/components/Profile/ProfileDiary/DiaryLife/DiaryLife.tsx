import STYLES from '../ProfileDiary.module.scss';

type Props = {
    maritalStatus: string | null;
    wife: string | null;
}

export function DiaryLife({ maritalStatus, wife }: Props) {
  if (maritalStatus !== null && wife !== null) {
    return (
      <>
        <div className={STYLES['fullwidth-main-card']}>
          Life
        </div>
        <div className={STYLES['halfwidth-cards-container']}>
          <div className={STYLES['halfwidth-secondary-card']}>
            <div className={STYLES['halfwidth-secondary-card-title']}><p>Marital Status</p></div>
            <div><span>{ maritalStatus }</span></div>
          </div>
          <div className={STYLES['halfwidth-secondary-card']}>
            <div className={STYLES['halfwidth-secondary-card-title']}><p>Wife</p></div>
            <div><span>{ wife }</span></div>
          </div>
        </div>
      </>
    );
  }

  if (maritalStatus !== null && wife === null) {
    return (
      <div className={STYLES['halfwidth-cards-container']}>
        <div className={STYLES['halfwidth-main-card']}>
          <span>Life</span>
        </div>
        <div className={STYLES['halfwidth-secondary-card']}>
          <div className={STYLES['halfwidth-secondary-card-title']}><p>Marital Status</p></div>
          <div><span>{ maritalStatus }</span></div>
        </div>
      </div>
    );
  }

  return null;
}
