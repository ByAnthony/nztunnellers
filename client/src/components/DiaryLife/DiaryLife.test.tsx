import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import { DiaryLife } from './DiaryLife';

const component = (
  <DiaryLife maritalStatus="Married" wife="Jane Doe" />
);

test('renders the component correctly', () => {
  const { asFragment } = render(component);

  expect(asFragment()).toMatchSnapshot();
});

test('renders life when known', () => {
  render(component);

  expect(screen.getByText('Life')).toBeInTheDocument();
  expect(screen.getByText('Marital Status')).toBeInTheDocument();
  expect(screen.getByText('Married')).toBeInTheDocument();
  expect(screen.getByText('Wife')).toBeInTheDocument();
  expect(screen.getByText('Jane Doe')).toBeInTheDocument();
});

describe.each(['Single', 'Widower', 'Separated'])('renders marital status if', (maritalStatus) => {
  test(`${maritalStatus}`, () => {
    const componentWithOtherMaritalStatus = (
      <DiaryLife maritalStatus={maritalStatus} wife={null} />
    );

    render(componentWithOtherMaritalStatus);

    expect(screen.getByText('Life')).toBeInTheDocument();
    expect(screen.getByText('Marital Status')).toBeInTheDocument();
    expect(screen.getByText(maritalStatus)).toBeInTheDocument();
    expect(screen.queryByText('Wife')).not.toBeInTheDocument();
    expect(screen.queryByText('Jane Doe')).not.toBeInTheDocument();
  });
});

test('does not render residence when unknown', () => {
  const { container } = render(
    <DiaryLife maritalStatus={null} wife={null} />,
  );

  expect(container).toBeEmptyDOMElement();
});
