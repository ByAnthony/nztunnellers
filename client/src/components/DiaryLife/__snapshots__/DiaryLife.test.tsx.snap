// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders the component correctly 1`] = `
<DocumentFragment>
  <div
    class="fullwidth-main-card"
  >
    Life
  </div>
  <div
    class="halfwidth-cards-container"
  >
    <div
      class="halfwidth-secondary-card"
    >
      <div
        class="halfwidth-secondary-card-title"
      >
        <p>
          Marital Status
        </p>
      </div>
      <div>
        <span>
          Married
        </span>
      </div>
    </div>
    <div
      class="halfwidth-secondary-card"
    >
      <div
        class="halfwidth-secondary-card-title"
      >
        <p>
          Wife
        </p>
      </div>
      <div>
        <span>
          Jane Doe
        </span>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
